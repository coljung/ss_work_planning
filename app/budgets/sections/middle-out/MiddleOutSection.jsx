import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Tabs } from 'antd';
import SectionContainer from '../SectionContainer';

// Sub Component
const TabPane = Tabs.TabPane;

export const TAB_BG_WOMEN = 'bg-women';
export const TAB_BG_MEN = 'bg-men';

export default class MiddleOutSection extends Component {
    constructor(props) {
        super(props);

        const { activeKey } = this.props;

        this.state = {
            activeTab: activeKey || TAB_BG_WOMEN,
            [TAB_BG_WOMEN]: false,
            [TAB_BG_MEN]: false,
        };

        this.handleTabChange = this.handleTabChange.bind(this);
    }

    handleTabChange(newTabKey) {
        // set true to load tabbed component
        const { activeKey } = this.props;

        this.setState({
            [activeKey]: false,
            activeTab: newTabKey,
            [newTabKey]: true,
        });

        this.props.changeTab(newTabKey);
    }

    render() {
        const { budget, version } = this.props;
        const { activeTab } = this.state;
        return (
            <Tabs activeKey={activeTab} onChange={this.handleTabChange} animated={false}>
                <TabPane tab="Brand Groups Women" key={TAB_BG_WOMEN}>
                    {(activeTab === TAB_BG_WOMEN || this.state[TAB_BG_WOMEN]) &&
                        <SectionContainer
                            budget={budget}
                            version={version}
                            key={TAB_BG_WOMEN}
                            view={TAB_BG_WOMEN}
                        />
                    }
                </TabPane>
                <TabPane tab="Brand Groups Men" key={TAB_BG_MEN}>
                    {(activeTab === TAB_BG_MEN) &&
                        <SectionContainer
                            budget={budget}
                            version={version}
                            key={TAB_BG_MEN}
                            view={TAB_BG_MEN}
                        />
                    }
                </TabPane>
            </Tabs>
        );
    }
}

MiddleOutSection.propTypes = {
    activeKey: PropTypes.string.isRequired,
    budget: PropTypes.string.isRequired,
    version: PropTypes.string.isRequired,
    changeTab: PropTypes.func.isRequired,
};
