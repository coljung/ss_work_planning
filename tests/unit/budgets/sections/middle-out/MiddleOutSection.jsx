import React from 'react';
import Enzyme, { mount } from 'enzyme';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MiddleOutSection from '../../../../../app/budgets/sections/middle-out/MiddleOutSection';

describe('MiddleOutSection', () => {
    it('should render base layout', () => {
        const output = shallow(
            <MiddleOutSection title="fsdfsdf">Hello Jest!</MiddleOutSection>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
