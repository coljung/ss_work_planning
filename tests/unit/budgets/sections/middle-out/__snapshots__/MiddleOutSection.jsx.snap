// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiddleOutSection should render base layout 1`] = `
<Tabs
  activeKey="bg-women"
  animated={false}
  hideAdd={false}
  onChange={[Function]}
  prefixCls="ant-tabs"
>
  <TabPane
    key="bg-women"
    placeholder={null}
    tab="Brand Groups Women"
  >
    <Connect(withRouter(SectionContainer))
      key="bg-women"
      view="bg-women"
    />
  </TabPane>
  <TabPane
    key="bg-men"
    placeholder={null}
    tab="Brand Groups Men"
  />
</Tabs>
`;
